packages/medusa-payment-stripe/src/services/stripe-provider.js

```diff
@@ -211,22 +203,23 @@ class StripeProviderService extends PaymentService {
    */
   async authorizePayment(sessionData, context = {}) {
     const stat = await this.getStatus(sessionData.data)
-
-    try {
-      return { data: sessionData.data, status: stat }
-    } catch (error) {
-      throw error
-    }
+    return { data: sessionData.data, status: stat }
   }
 
   async updatePaymentData(sessionData, update) {
-    try {
-      return this.stripe_.paymentIntents.update(sessionData.id, {
-        ...update.data,
-      })
-    } catch (error) {
-      throw error
+    const { payment_method, ...rest } = update
```

Comment:
removed a `.data` here because it seemed to be doubly nested.  

https://github.com/medusajs/medusa/blob/master/packages/medusa/src/api/routes/store/carts/update-payment-session.ts#L38 already fetches the `.data`

Reply:
This one feels a bit dangerous - can you explain a bit more about what is going on :)

Reply:
Sure! I explained in the PR description, but I'll expand more here.

_fix(stripe): doubly nested data prop when updating payment data_

This is what listing payment methods looks like via the storefront list payment methods API:

```
    "payment_methods": [
      {
        "provider_id": "stripe",
        "data": {
          "id": "pm_1JumgWJa5dwDH87yhp4oxcV7",
          "object": "payment_method",
          "card": {
            "brand": "visa",
            "country": "US"
            ...
          }
        }
      },
```

So you would think that to use the update-payment-session admin api, you should pass stripe props inside the data prop that it expects. But that's not the case. I had to double nest the stripe props inside 2 data props like this: 

```
const response = await this.axiosClient.post<{ cart: Cart }>(`/carts/${cartId}/payment-sessions/${provider_id}`, 
{ data: { data: { payment_method: paymentMethod.data.id } } });
```
So I'm thinking the issue is here, doing this seemed to fix the issue.

It could also be that this line is incorrect here, where it's passing the `.data` prop on to the payment service, instead of the full body.
https://github.com/medusajs/medusa/blob/master/packages/medusa/src/api/routes/store/carts/update-payment-session.ts#L38

I'm not really sure, it's hard to know what types are expected where right now.

Reply:
Also see the PR description for why I added stripe `paymentIntents.confirm` for the payment method below.